build.gradle

```diff
@@ -73,6 +83,22 @@ subprojects {
   apply plugin: 'nebula.dependency-recommender'
   apply plugin: 'java-library'
 
+  // apply this plugin in a try-catch block so that we can handle cases without .git directory
+  // apply to subprojects so that git properties information is available for runtime debugging and tests
+  //  e.g. sending build SHA as headers
+  try {
+    apply plugin: 'com.gorylenko.gradle-git-properties'
+    gitProperties {
+      gitPropertiesName = "${project.name}-version.properties"
+      extProperty = 'gitProps'
+      failOnNoGitDirectory = false
```

Comment:
nit: I wonder if the try-catch is necessary with `failOnNoGitDirectory=false`. Did you have a chance to try this without a .git directory?

Reply:
Yeah it's still necessary.

Reply:
I tried removing the `.git` directory entirely using `git clone --depth=1 --branch=kb-gradle-git-properties-plugin-file-generation git@github.com:kbendick/iceberg.git iceberg-version-no-git && rm -rf .git` and the build errors out because `gitProperties` isn't recognized (because the plugin apply didn't work).

So I'm assuming that property is more for cases where there's no git directory, but JGit is still available.